core/src/main/java/io/undertow/conduits/ChunkedStreamSinkConduit.java

```diff
@@ -128,6 +129,108 @@ public int write(final ByteBuffer src) throws IOException {
         return doWrite(src);
     }
 
+    long doWrite(final ByteBuffer[] srcs, int offset, int length) throws IOException {
+        if (anyAreSet(state, FLAG_WRITES_SHUTDOWN)) {
+            throw new ClosedChannelException();
+        }
+        long totalRemaining = Buffers.remaining(srcs, offset, length);
+        if(totalRemaining == 0) {
+            return 0;
+        }
+        if (totalRemaining > Integer.MAX_VALUE) {
+            // Fall back to one buffer at a time if the total srcs remaining exceeds integer max-value
+            // This should be very rare.
+            for (int i = offset; i < offset + length; i++) {
+                ByteBuffer buf = srcs[i];
+                if (buf.hasRemaining()) {
+                    return write(buf);
+                }
+            }
+            return 0;
```

Comment:
In this case, we could internally reduce `length` and potentially dial back the limit on the final buffer, matching the chunkleft>0 case on 170